### ISSUE TITLE
Issues with tooltips with multiple displays with different DPIs

### ISSUE THREAD

**Issue by blin4444** [amcharts/amcharts4#676]
I have an issue where graph tooltips in an XYChart display correctly only on one display (incidentally, the low DPI one). Reloading the page with the browser showing in different displays does not help.
Displays:
- 24" DELL U2415 at 1920×1200: tooltips shown correctly
- MacBook Pro 13" internal display at 2560×1600 (macOS 10.14, setting in Display Preferences: More Space): tooltips not shown correctly

Even when the external display is not connected, the tooltips do not display correctly on the internal display, no matter which scale preference is selected in Display Preferences (More Space, Standard, etc.)

Code to set tooltip content:
```
var newSeries = chart.series.push(new am4charts.LineSeries());
// ...
newSeries.tooltipHTML = myTimeSeries.sensor.id + ' <strong>' + myTimeSeries.attribute.displayName + '</strong>: {valueY.value}';
styleTooltip(newSeries);
```

Code to customize tooltip:
```
function styleTooltip(obj) {
  obj.tooltip.background.fill = am4core.color('#000000');
  obj.tooltip.background.cornerRadius = 0;
  obj.tooltip.background.fillOpacity = AmChartManager.TOOLTIP_FILL_OPACITY;
  obj.tooltip.background.strokeOpacity = 0;
  obj.tooltip.background.filters.clear();
  obj.tooltip.filters.clear();
  obj.tooltip.animationDuration = 500;
  obj.tooltip.hiddenState.transitionDuration = 500;
}
```

Safari on notebook display:
<img width="278" alt="screen shot 2018-12-12 at 2 56 38 pm" src="https://user-images.githubusercontent.com/2326556/49904580-737fd980-fe1f-11e8-80c4-50a5e1179ac9.png">

Chrome on notebook display:
<img width="254" alt="screen shot 2018-12-12 at 3 00 19 pm" src="https://user-images.githubusercontent.com/2326556/49904582-737fd980-fe1f-11e8-854c-f8648a7a0635.png">

External monitor:
![screen shot 2018-12-12 at 2 56 48 pm](https://user-images.githubusercontent.com/2326556/49904642-a4600e80-fe1f-11e8-9657-42cd26b278dd.png)

Seems like it could be an issue with setting tooltipHTML. Minimal example: https://codepen.io/blin4444/pen/MZarPg

**Comment by blin4444** [comment]
Same issue: https://github.com/amcharts/amcharts4/issues/684

**Comment by martynasma** [comment]
Must be Safari thing. Thanks for the heads up. Testing.

**Comment by blin4444** [comment]
It is also happening in Chrome and Firefox.

**Comment by mfamilia** [comment]
Debugging in Safari shows that `foreignObject` used for tooltip is exactly have the width.

If you look at the Label element code, there is one place where the width is halved,

https://github.com/amcharts/amcharts4/blob/master/src/.internal/core/elements/Label.ts#L1007

Removing the division operation, fixes our issue (#684) but not sure what cases would break because of it.

**Comment by martynasma** [comment]
Well, that division is there for a reason. Without it the tooltips would be twice as large on other browsers.

I suppose there are differences on how whether how Safari and other browsers calculate bounding box. Looks like Safari already factors pixelRatio in. We're looking for ways to detect that.

Will keep you posted.

**Comment by martynasma** [comment]
This issue was fixed in dev version, and will be released soon.

We'll notify you here when that happens.

**Comment by blin4444** [comment]
Alright, thanks!

**Comment by martynasma** [comment]
Fixed in [4.0.9](https://github.com/amcharts/amcharts4/releases/tag/4.0.9).

## [4.0.9] - 2018-12-19

### Added
- `snapToSeries` added to `XYCursor`. Allows setting series to which cursor lines should be snapped. Works if one of the axis of the series is `DateAxis` or `CategoryAxis`. Won't work if both axes are `ValueAxis`.
- New example: Pie charts in Columns using JSON config.
- Export: `emptyAs` added to both CSV and Excel export options. If set, missing values will be replaced with this value (default is empty string).
- New event `"beforedisposed"` added to `Sprite`. Kicks in right before the element starts to dismantle itself.
- Italian translation.

### Changed
- Export: if no `dataFields` specify it will now look for fields in all of the data, not just the first line.
- When creating a new state on an object, it will not automatically propagate itself on object's clones if `applyOnClones = true`.
- Setting `Container`'s `setStateOnChildren` will now propage to the new value to its existing clones if `applyOnClones = true`.
- In case `deepInvalidate()` is called on `Container`, it forces labels to redraw.
- `filters` on `SpriteState` is now `List` (was `ListTemplate`). It's better for performance which does not require creating Filter template object.

### Fixed
- Clicking a hoverable/clickable object was resultin in "over" and "out" events generated next to "hit".
- JSON: properties to existing `List` items were not being applied properly.
- Using geodata in Ember app was causing errors (Issue 672).
- Export to CSV/Excel was following first line of data values. It now respects `dataFields` setting, and will replace missing values with `emptyAs` (new options setting).
- Wheel-scrolling in MacOS Safari was very slow.
- HTML-based tooltips were too small to fit the actual contents on MaCOS Safari.
- `DateFormatter` was not parsing `"i"` format as local timezone, rather than UTC.
- Sometimes bullets with `Rectangle` element in them were not positioned properly.
- If category was named `"undefined"`, it was messing up label positions on `CategoryAxis`.
- State's property value was not being applied if current value of the property was `undefined`.
- Performance of `PieChart` and `RadarChart` improved dramatically under IEs.
- In some cases `"ready"` event was not being fired.
- `FunnelSeries` with 0 (zero) value slices was not rendered correctly.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
